src/vconsole/vconsole-setup.c

```diff
@@ -537,28 +544,33 @@ static int find_source_vc(char **ret_path, unsigned *ret_idx) {
 
                 fd = open_terminal(path, O_RDWR|O_CLOEXEC|O_NOCTTY);
                 if (fd < 0) {
-                        RET_GATHER(err, log_debug_errno(fd, "Failed to open terminal %s, ignoring: %m", path));
+                        log_debug_errno(fd, "Failed to open terminal %s, ignoring: %m", path);
+                        if (IN_SET(err, 0, -EBUSY, -ENOENT))
+                                err = fd;
                         continue;
                 }
 
                 r = verify_vc_kbmode(fd);
+                if (r >= 0)
+                        r = verify_vc_display_mode(fd);
                 if (r < 0) {
-                        RET_GATHER(err, log_debug_errno(r, "Failed to check VC %s keyboard mode: %m", path));
+                        log_debug_errno(r, "Failed to check VC %s keyboard or display mode: %m", path);
+                        if (IN_SET(err, 0, -ENOENT))
+                                err = r;
                         continue;
                 }
 
-                r = verify_vc_display_mode(fd);
-                if (r < 0) {
-                        RET_GATHER(err, log_debug_errno(r, "Failed to check VC %s display mode: %m", path));
-                        continue;
-                }
+                log_debug("Selecting %s as source console", path);
 
                 /* all checks passed, return this one as a source console */
                 *ret_idx = i;
                 *ret_path = TAKE_PTR(path);
                 return TAKE_FD(fd);
         }
 
+        if (err == -EBUSY)
+                return 0;
```

Comment:
in systemd we have this rule that all function ret params must be left untouched on failure, and fully initialized on success. returning 0 is a form of success, hence please initialize the return params to something

Reply:
honestly this looks weird to me. It seems to me  that the function should just consider "all VCs are busy" as an error and let the caller deals with the specific case (-EBUSY).

Reply:
So I took the liberty to implement that, if you still think that the function should return 0 and it only found busy VCs, I'll update this part.

Reply:
returning EBUSY in this case is OK, as long as the function doesn't log above LOG_DEBUG about any errors, and leaves that to the caller. But that's given here, hence what you did is fine